### PR TITLE
Replacing all occurences of 't8_topidx_t' definitions.

### PR DESCRIPTION
**Pull request by jmark**
**_Describe your changes here:_**

Replacing or removing all occurences of 't8_topidx_t' definitions. As simple as that.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

**Review by holke — CHANGES_REQUESTED** [review]
Great, thanks for resolving this.

Only one minor change left, reverting p4est_topidx removal.

File: src/t8.h [hunk 1]
```diff
@@ -64,31 +63,52 @@ T8_EXTERN_C_BEGIN ();
...
 
-#define T8_ASSERT P4EST_ASSERT          /**< TODO: write proper function. */
-#define T8_ALLOC P4EST_ALLOC            /**< TODO: write proper function. */
-#define T8_ALLOC_ZERO P4EST_ALLOC_ZERO  /**< TODO: write proper function. */
-#define T8_FREE P4EST_FREE              /**< TODO: write proper function. */
-#define T8_REALLOC P4EST_REALLOC        /**< TODO: write proper function. */
+/** Widely used assertion. Only active in debug-mode. */
+/* Note that we use the same definition as in sc.h. We are deliberately not using
+ * #define T8_ASSERT SC_ASSERT
+ * since then the assertion would not trigger if sc is not configure in debugging mode.
```
Comment (holke) [inline src/t8.h:69]:
```suggestion
 * since then the assertion would not trigger if sc is not configured in debugging mode.
```

File: src/t8.h [hunk 2]
```diff
@@ -64,31 +63,52 @@ T8_EXTERN_C_BEGIN ();
...
+ * However, we want it to trigger anytime t8code is in debugging mode, independent of sc.
```
Comment (holke) [inline src/t8.h:70]:
```suggestion
 * However, we want it to trigger any time t8code is in debugging mode, independent of sc.
```

File: src/t8_cmesh/t8_cmesh_examples.c [hunk 3]
```diff
@@ -54,6 +54,10 @@ t8_cmesh_new_from_p4est_ext (void *conn, int dim,
   p4est_topidx_t      ttt;
   t8_geometry_c      *linear_geom = t8_geometry_linear_new (3);
 
+  /* Make sure that p4est is properly initialized. If not, do it here. */
+  if (!sc_package_is_registered (p4est_package_id))
+    p4est_init (NULL, SC_LP_ESSENTIAL);
+
```
Comment (holke) [inline src/t8_cmesh/t8_cmesh_examples.c:60]: Please add our discussed changes: A warning via t8_global_errorf and a note in the doxy comment in the function declaration.

**Review by holke — CHANGES_REQUESTED** [review]
Great, only 1 change left.

File: src/t8_cmesh/t8_cmesh_examples.c (hunk 3 above)
Comment (jmark) [reply src/t8_cmesh/t8_cmesh_examples.c:63]: Done.

**Review by holke — APPROVED** [review]

**Pull request merged**